### PR TITLE
Properly cleanup and setup APY tests

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

Properly cleanup and setup the APY tests after the `setup` and `setup_class` modifications.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

File: packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py [hunk 1]
```diff
@@ -277,6 +278,7 @@ def test_current_pair_ids(
         behaviour.params.pair_ids = pairs_ids
 
         assert expected_pair_ids == behaviour.current_pair_ids
+        self.end_round()
```
Comment (Adamantios) [inline packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py:281]: Ending round after all tests.

File: packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py [hunk 2]
```diff
@@ -407,6 +413,8 @@ def test_retries_exceeded(
         # we expect `retries_exceeded` to return True if any of the subgraphs' `is_retries_exceeded` returns `True`
         expected = any(is_exceeded_per_subgraph)
         assert behaviour.retries_exceeded == expected
+        behaviour._utilized_subgraphs = {}
```
Comment (Adamantios) [inline packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py:416]: Reset `_utilized_subgraphs` after the check.

File: packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py [hunk 3]
```diff
@@ -800,6 +813,10 @@ def test_invalid_pairs(self, caplog: LogCaptureFixture) -> None:
 
         self.mock_a2a_transaction()
         self._test_done_flag_set()
+
+        behaviour._progress.call_failed = False
+        behaviour.context.spooky_subgraph._retries_attempted = 0
```
Comment (Adamantios) [inline packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py:818]: Reset flag and retries before ending the round.

File: packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py [hunk 4]
```diff
@@ -825,6 +842,10 @@ def test_fetch_behaviour_non_indexed_block(
         # we do this because of https://github.com/valory-xyz/open-autonomy/pull/646
         behaviour._check_given_pairs = mock.MagicMock()  # type: ignore
         behaviour._pairs_exist = True
+        behaviour._progress.initialized = True
+        behaviour._progress.current_dex_name = "uniswap_subgraph"
+        expected_timestamp = 1
+        behaviour._progress.timestamps_iterator = iter((expected_timestamp,))
```
Comment (Adamantios) [inline packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py:848]: Set the progress properly for the test.

File: packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py [hunk 5]
```diff
@@ -942,6 +965,12 @@ def test_fetch_behaviour_non_indexed_block_none_res(
         # we do this because of https://github.com/valory-xyz/open-autonomy/pull/646
         behaviour._check_given_pairs = mock.MagicMock()  # type: ignore
         behaviour._pairs_exist = True
+        behaviour._progress.initialized = True
+        behaviour._progress.current_dex_name = "uniswap_subgraph"
+        expected_timestamp = 1
+        behaviour._progress.timestamps_iterator = (
+            iter((expected_timestamp,)) if expected_timestamp is not None else iter(())
+        )
```
Comment (Adamantios) [inline packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py:973]: Set the progress properly for the test.

File: packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py [hunk 6]
```diff
@@ -1114,6 +1147,12 @@ def test_fetch_value_none(
...
```
Comment (Adamantios) [inline packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py:1155]: Set the progress properly for the test.

File: packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py (hunk 1 above)
Comment (DavidMinarsch) [reply packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py:281]: you can also do this in `teardown` next time. Would be good to collect several refactoring opportunities at once though.

File: packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py (hunk 2 above)
Comment (DavidMinarsch) [reply packages/valory/skills/apy_estimation_abci/tests/test_behaviours.py:416]: Also, generally would do this on the teardown or setup so it's very easy to see all the conditions in one place.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**